librz/core/cmd/cmd_analysis.c

```diff
@@ -373,98 +373,50 @@ static void list_vars(RzCore *core, RzAnalysisFunction *fcn, PJ *pj, int type, c
 	}
 }
 
-static bool core_analysis_name(RzCore *core, const char *name, PJ *pj) {
-	bool ret = true;
-	ut64 off = core->offset;
-	RzAnalysisOp op;
-	ut64 tgt_addr = UT64_MAX;
-
-	if (pj) {
-		pj_a(pj);
+#define PJ_KS(pj, key, value) \
+	{ \
+		const char *value_tmp = (value); \
+		if (RZ_STR_ISNOTEMPTY(value_tmp)) { \
+			pj_ks(pj, key, value_tmp); \
```

Comment:
@thestr4ng3r @XVilka i wonder whether we should just check in the `pj` API whether the value is present or not.. WDYT? For pj_kn maybe we can have a separate API to avoid assuming UT64_MAX as "empty value".

Reply:
Maybe